clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/util/id/DcosSpinnakerAppId.groovy

```diff
@@ -0,0 +1,153 @@
+package com.netflix.spinnaker.clouddriver.dcos.deploy.util.id
+
+import static com.google.common.base.Preconditions.checkArgument
+import static com.google.common.base.Strings.nullToEmpty
+
+import com.netflix.frigga.Names
+
+/**
+ * Represents a hierarchical Spinnaker specific application identifier for DCOS.
+ * Structure - /account/region/app-stack-detail-sequence
+ */
+class DcosSpinnakerAppId {
+    public final static def SAFE_REGION_SEPARATOR = "_"
+
+    private final def marathonPath
+
+    public DcosSpinnakerAppId(final String marathonAppId) {
```

Comment:
Instead of public constructors, can we maintain the static constructors we used to have? I found them to be much more descriptive about the creation of the app id (i.e. DcosSpinnakerAppId.parse(...) meant we were parsing from a string, DcosSpinnakerAppId.from(...) meant we were creating from known fields, etc.